src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -6,17 +6,24 @@
 using System.Diagnostics;
 using Microsoft.CodeAnalysis.CSharp.Symbols;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     internal partial class Binder
     {
+        /// <summary>
+        /// constants used to designate escaping scopes.
```

Comment:
Consider making an `enum` for this so the use sites aren't typed as uint everywhere.

Reply:
these are not the only values. It is actually a uint. Just 0 and 1 are special, so I made them constants. 
The rest is just how deeply the scope is nested - each level adds 1

---
In reply to: [135153899](https://github.com/dotnet/roslyn/pull/21334#discussion_r135153899) [](ancestors = 135153899)

Reply:
Also added a comment explaining how scopes are numbered.

---
In reply to: [135157848](https://github.com/dotnet/roslyn/pull/21334#discussion_r135157848) [](ancestors = 135157848,135153899)